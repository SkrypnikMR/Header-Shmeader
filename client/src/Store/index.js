import { combineReducers } from 'redux';
import { reducer as chatReducer } from './chat/reducer';
import { reducer as registrationReducer } from './registration/reducer';
import { reducer as loginReducer } from './login/reducer';
import { reducer as userReducer } from './user/reducer';

const rootReducer = combineReducers({
  chat: chatReducer,
  registration: registrationReducer,
  login: loginReducer,
  user: userReducer,
});


export default rootReducer;
