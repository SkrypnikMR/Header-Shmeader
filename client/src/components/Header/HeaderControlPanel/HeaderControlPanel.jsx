import React from 'react';
import { useTranslation } from 'react-i18next';
import { StControl } from './styled';
import { HEADER_CONTROL_BTNS } from '/src/constants/componentsÐ¡onsts.js';
import Button from '../../UI/Button';

const HeaderControlPanel = () => {
    const { i18n } = useTranslation();
    const handleClick = (e) => {
        i18n.changeLanguage(e.target.value);
        localStorage.setItem('lang', e.target.value);
    };
    const handleThemeClick = () => {
     
   };
    return (
        <StControl >
            {HEADER_CONTROL_BTNS.map(el => (
                <Button
                    id={el.id}
                    content={el.content}
                    key={el.id}
                    color='white'
                    width='60px'
                    height="10vh"
                    borderRadius="0px"
                    value={el.id}
                    bgColor='rgba(0,0,0,0)'
                    onClick={el.id === el.theme_btn ? handleClick : handleThemeClick}
                />
            ))}
        </StControl>
    );
};

export default HeaderControlPanel;
