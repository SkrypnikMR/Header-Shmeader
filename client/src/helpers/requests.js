export const getRequest = async (url, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = { method: 'GET', headers };
  const answer = await fetch(url, options);
  return answer.json();
};
export const postRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'POST',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(url, options);
  return answer.json();
};

export const putRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'PUT',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(url, options);
  return answer.json();
};
export const createAuthHeader = (support) => { return { Authorization: support.getCookie('token') }; };
