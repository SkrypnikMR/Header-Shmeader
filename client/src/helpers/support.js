export const support = {
    setSessionStorageItem: (name, data) => {
        if (typeof data !== 'string') data = JSON.stringify(data);
        sessionStorage.setItem(name, data);
    },
    killSessionStorageItem: (name) => {
        sessionStorage.removeItem(name);
    },
    getSessionStorageItem: (name) => {
        const item = sessionStorage.getItem(name);
        return item?.charAt(0) === '{' ? JSON.parse(item) : item;
    },
    getPrettyTime: time => `${new Date(Number(time)).toTimeString().substr(0, 8)}`,
    getMessagesFolders: (rooms) => {
        const messagesFolders = {};
        rooms.forEach((room) => {
            const { room_name } = room;
            messagesFolders[room_name] = [];
        });
        return messagesFolders;
    },
    filteredRooms: (filterValue, rooms) => {
        const result = rooms.filter(room => room.room_name.includes(filterValue));
        return result.length === 0 ? null : result;
    },
};
