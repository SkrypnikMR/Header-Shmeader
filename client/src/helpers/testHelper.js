import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

export const shallowSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    const test = <Router>{core}</Router>;
    return shallow(test);
};
