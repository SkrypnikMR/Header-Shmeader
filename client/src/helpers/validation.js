import { regExp } from '../constants/regExp';

export const validation = {
    registrationValidation: ({ email, password, confirm, firstName, lastName }) => {
        let answer = { message: '', isValid: true };

        if (!regExp.emailRegExp.test(email) || !email) answer = { message: 'Invalid email', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) answer = { message: 'Invalid password', isValid: false };
        if (!regExp.firstNameRegExp.test(firstName) || !firstName) answer = { message: 'Invalid firstName', isValid: false };
        if (!regExp.lastNameRegExp.test(lastName) || !lastName) answer = { message: 'Invalid lastName', isValid: false };
        if (password !== confirm) answer = { message: 'Password mismatch', isValid: false };
        return answer;

    },
    loginValidation: ({ email, password }) => {
        let answer = { message: '', isValid: true };

        if (!regExp.emailRegExp.test(email) || !email) answer = { message: 'Invalid email', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) answer = { message: 'Invalid password', isValid: false };
        return answer;
    },
};
