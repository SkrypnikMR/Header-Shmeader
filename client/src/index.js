import '../public/styles/styles.scss';
import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import rootReducer from './Store';
import App from './components/App';
import rootSaga from './Store/saga';
import './i18n';

/* const devTools = window?.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(); devTools, */
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(saga),
  ),
);
saga.run(rootSaga);

render(
  <Provider store={store}>
    <App />
  </Provider >,
  document.getElementById('root'),
);
