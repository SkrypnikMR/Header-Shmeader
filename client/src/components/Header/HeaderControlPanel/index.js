import { withRouter } from 'react-router';
import HeaderControlPanel from './HeaderControlPanel.jsx';
import { connect } from 'react-redux';
import { setValue } from '/src/Store/user/actions';
import { userThemeMode } from '/src/Store/user/selectors';

const mapStateToProps = state => ({ themeMode: userThemeMode(state) });
const mapDispatchToProps = dispatch => ({
    setValue: payload => dispatch(setValue(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(HeaderControlPanel));
