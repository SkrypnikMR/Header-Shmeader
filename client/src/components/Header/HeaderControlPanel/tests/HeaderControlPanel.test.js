import React from 'react';
import { shallowSmart, mountSmart } from '/src/helpers/testHelper';
import HeaderControlPanel from '../HeaderControlPanel';
import '/src/i18n';

describe('App', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<HeaderControlPanel />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render inputs', () => {
        const component = mountSmart(<HeaderControlPanel />);
        expect(component.find('button')).toHaveLength(3);
    });
});
