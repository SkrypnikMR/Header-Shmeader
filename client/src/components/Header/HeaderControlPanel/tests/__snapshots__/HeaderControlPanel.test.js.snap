// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `"<div class=\\"styled__StControl-sc-1i99xqk-0 joBAxQ\\"><button id=\\"en\\" type=\\"button\\" color=\\"white\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 iwKgks\\">EN</button><button id=\\"ru\\" type=\\"button\\" color=\\"white\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 iwKgks\\">RU</button><button id=\\"ar\\" type=\\"button\\" color=\\"white\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 iwKgks\\">AR</button></div>"`;
